### PR TITLE
Update samples to Alpine

### PR DESCRIPTION
**Pull request by ashnaga**
Fixes https://github.com/dotnet/dotnet-docker/issues/2167

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/dotnet-docker?pullRequest=4171) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/dotnet-docker?pullRequest=4171) before we can accept your contribution.<br/><br/>:x: ashnaga [sign now](https://cla.dotnetfoundation.org/dotnet/dotnet-docker?pullRequest=4171)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/dotnet-docker?pullRequest=4171) it.</sub>

**Comment by mthalman** [comment]
Because the tag names have changed, you'll also need to update this file to reference those new names: https://github.com/dotnet/dotnet-docker/blob/main/eng/mcr-tags-metadata-templates/samples-tags.yml. After updating that file, run `.\eng\readme-templates\Get-GeneratedReadmes.ps1` to update the readme to reflect the new tags.

File: samples/aspnetapp/Dockerfile.alpine-slim [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+# copy everything else and build app
+COPY aspnetapp/. .
+RUN dotnet publish -c Release -o /app --use-current-runtime --self-contained true --no-restore /p:PublishTrimmed=true /p:PublishSingleFile=true
+
+# final stage/image
+FROM mcr.microsoft.com/dotnet/runtime-deps:7.0-alpine
+WORKDIR /app
+COPY --from=build /app .
+ENTRYPOINT ["./aspnetapp"]
```
Comment (mthalman) [inline samples/aspnetapp/Dockerfile.alpine-slim:17]: Please add a newline at the end of the file. Applies to the other Dockerfile too.

Comment (ashnaga) [reply samples/aspnetapp/Dockerfile.alpine-slim]: Done

**Comment by ashnaga** [comment]
> Because the tag names have changed, you'll also need to update this file to reference those new names: https://github.com/dotnet/dotnet-docker/blob/main/eng/mcr-tags-metadata-templates/samples-tags.yml. After updating that file, run `.\eng\readme-templates\Get-GeneratedReadmes.ps1` to update the readme to reflect the new tags.

Done

**Comment by mthalman** [comment]
FYI: I rebased this to nightly so we can get it merged there first. Then we'll merge it to main and get the new images published as part of the November release.

**Review by mthalman — APPROVED** [review]

**Review by MichaelSimons — APPROVED** [review]

**Comment by mthalman** [comment]
I'm investigating an issue with the tests where they're not testing what we expect to be tested. This was apparently an issue prior to these changes but I just discovered it today.

**Comment by mthalman** [comment]
I have fixed the tests locally but that's uncovered a product issue which I've logged at https://github.com/dotnet/sdk/issues/28809.

**Comment by mthalman** [comment]
> I have fixed the tests locally but that's uncovered a product issue which I've logged at [dotnet/sdk#28809](https://github.com/dotnet/sdk/issues/28809).

Looks like this won't be fixed in GA. Our choices:

1. Hold off on making these changes until it is fixed (hopefully 7.0.1)
2. Get rid of the Dockerfile currently defined in this PR and replace it with Dockerfile.alpine-arm64-slim and Dockerfile.alpine-arm32-slim (we already have Dockerfile.alpine-x64-slim). Then when it is fixed we can add the multi-arch Dockerfile.alpine-slim file and probably delete those two Arm-based Dockerfiles.

**Comment by ashnaga** [comment]
> 2\. Get rid of the Dockerfile currently defined in this PR and replace it with Dockerfile.alpine-arm64-slim and Dockerfile.alpine-arm32-slim (we already have Dockerfile.alpine-x64-slim). Then when it is fixed we can add the multi-arch Dockerfile.alpine-slim file and probably delete those two Arm-based Dockerfiles.

We should pick the option 2 as the goal is to have reduced size samples (using arch specific Dockerfiles) for better customer experience. But I am not sure on the benefit of deleting and adding the multi-arch file later.

**Comment by mthalman** [comment]
> We should pick the option 2 as the goal is to have reduced size samples (using arch specific Dockerfiles) for better customer experience.

Option 1 gives us the reduced size as well, just a month later.

> But I am not sure on the benefit of deleting and adding the multi-arch file later.

The multi-arch file is the canonical and more maintainable solution that we would want to guide customers to use.

**Comment by MichaelSimons** [comment]
I am in favor of #2.  There will be marketing buzz at 7.0.0 GA that we should take advantage of by making our published samples  quicker to pull therefore leaving a more favorable impression.

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**